### PR TITLE
GUVNOR-2811: Project Editor: Build button is not re-enabled if the user visits an empty Organizational Unit

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2811

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java [hunk 1]
```diff
@@ -431,10 +431,11 @@ public void onClose() {
 
     private void update() {
         if ( workbenchContext.getActiveProject() == null ) {
-            disableMenus();
+            enableMenus( false );
             view.showNoProjectSelected();
             view.hideBusyIndicator();
         } else {
+            enableMenus( true );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java]: This is the fix! The rest is code formatting.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java [hunk 2]
```diff
@@ -637,18 +638,22 @@ public void push( MenuFactory.CustomMenuBuilder element ) {
                     @Override
                     public MenuItem build() {
                         return new BaseMenuCustom<IsWidget>() {
+
+                            private boolean enabled;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java]: Enabled/disabled "state" was moved to an instance variable (and out of the ```Widget```) for Unit Tests.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/test/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenterTest.java [hunk 3]
```diff
@@ -590,6 +591,32 @@ public void testGetReimportCommand() throws Exception {
         verify( projectScreenService, times( 1 ) ).reImport( eq( presenter.pathToPomXML ) );
     }
 
+    @Test
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/test/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenterTest.java]: New tests; the rest is code formatting.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by csadilek** [comment]
+1

**Pull request merged**